server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -791,6 +791,27 @@ public void run() {
       }
     }, 0, 5000, TimeUnit.MILLISECONDS);
 
+    ThreadPools.createGeneralScheduledExecutorService(aconf).scheduleWithFixedDelay(() -> {
+      final SortedMap<KeyExtent,Tablet> onlineTabletsSnapshot = onlineTablets.snapshot();
+      final SortedMap<KeyExtent,Pair<Long,TabletMetadata>> tabletValidationInfo = new TreeMap<>();
+
+      // gather update counters and metadata for all tablets
+      for (var entry : onlineTabletsSnapshot.entrySet()) {
+        KeyExtent keyExtent = entry.getKey();
+        Long counter = entry.getValue().getUpdateCounter();
+        TabletMetadata tm = getContext().getAmple().readTablet(keyExtent,
+            TabletMetadata.ColumnType.FILES, TabletMetadata.ColumnType.LOGS,
+            TabletMetadata.ColumnType.ECOMP, TabletMetadata.ColumnType.PREV_ROW);
```

Comment:
Just thought of one wrinkle with using readTablets(), it may not support reading different data levels at the same time. Like it may not support reading from the root tablet and a user tablet in the same call.  I am not sure though.

Reply:
The most recent changes (4f282f4) use readTablets(). However, I am not familiar with this method so I'm not sure the difference this might make regarding the data levels. @keith-turner

Reply:
I looked and I think it does only support user tablets.

https://github.com/apache/accumulo/blob/6cfb9180a0d3e5115922314ff2062e0706ef0795/core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletsMetadata.java#L324-L328

So if the tablet server was serving the root or metadata tablets, then it would throw an exception for this.  Could partition the tablets into user and non-user (root,metadata) tablets.  Can use readTablets() w/ the user tablets.  For the non-user would have to call readTablet() one at a time.  Could do that partitioning in the tablet server code for now and in a later PR push it down into Ample and remove it from the tablet server code.

Reply:
@keith-turner, in 4f282f4 I created two subsets of tablets one with DataLevel User and another with the remaining tablets, then used `readTablets()` on the first set and `readTablet()` on the second. I also left a TODO regarding moving the functionality to ample and can follow up with a new ticket for that once this PR gets merged.

Reply:
The partitioning looks good.